src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Samples.cshtml.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using ApiView;
+using APIView;
+using APIViewWeb.Models;
+using APIViewWeb.Repositories;
+using Markdig.Helpers;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Cosmos.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json.Linq;
+
+namespace APIViewWeb.Pages.Assemblies
+{
+    public class UsageSamplePageModel : PageModel
+    {
+        private readonly UsageSampleManager _samplesManager;
+        private readonly ReviewManager _reviewManager;
+        private const string ENDPOINT_SETTING = "Endpoint";
+        private readonly CommentsManager _commentsManager;
+        private readonly NotificationManager _notificationManager;
+        public readonly UserPreferenceCache _preferenceCache;
+
+        public string Endpoint { get; }
+        public ReviewModel Review { get; private set; }
+        public UsageSampleModel Sample { get; private set; }
+        public IEnumerable<UsageSampleModel> SampleRevisions { get; private set; }
+        public CodeLineModel[] SampleContent { get; set; }
+        public ReviewCommentsModel Comments { get; set; }
+        public string SampleOriginal { get; set; }
+        public int latestRevision { get; set; }
+
+        public UsageSamplePageModel(
+            IConfiguration configuration,
+            UsageSampleManager samplesManager,
+            ReviewManager reviewManager,
+            CommentsManager commentsManager,
+            NotificationManager notificationManager,
+            UserPreferenceCache preferenceCache)
+        {
+            _samplesManager = samplesManager;
+            _reviewManager = reviewManager;
+            Endpoint = configuration.GetValue<string>(ENDPOINT_SETTING);
+            _commentsManager = commentsManager;
+            _notificationManager = notificationManager;
+            _preferenceCache = preferenceCache;
+        }
+
+        [FromForm]
+        public UsageSampleUploadModel Upload { get; set; }
+
+        public async Task<IActionResult> OnGetAsync(string id, string revisionId = null)
+        {
+            TempData["Page"] = "samples";
+            Review = await _reviewManager.GetReviewAsync(User, id);
+            Comments = await _commentsManager.GetUsageSampleCommentsAsync(id);
+            latestRevision = -1;
+
```

Comment:
We need to make sure we check the authorization of user here to avoid unauthorized access of this API. You can refer this assert check for an eg. https://github.com/Azure/azure-sdk-tools/blob/main/src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs#L345